engine/src/main/java/org/terasology/rendering/opengl/fbms/DisplayResolutionDependentFBOs.java

```diff
@@ -18,31 +18,46 @@
 import org.lwjgl.opengl.Display;
 import org.terasology.config.RenderingConfig;
 import org.terasology.engine.SimpleUri;
+import org.terasology.engine.subsystem.DisplayDevice;
+import org.terasology.registry.CoreRegistry;
 import org.terasology.rendering.opengl.AbstractFBOsManager;
 import org.terasology.rendering.opengl.FBO;
 import org.terasology.rendering.opengl.FBOConfig;
 import org.terasology.rendering.opengl.ScreenGrabber;
 import org.terasology.rendering.opengl.SwappableFBO;
 
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+import static org.terasology.config.RenderingConfig.FBO_SCALE;
+import static org.terasology.engine.subsystem.lwjgl.LwjglDisplayDevice.DISPLAY_RESOLUTION_CHANGE;
 import static org.terasology.rendering.opengl.ScalingFactors.FULL_SCALE;
 
 /**
  * TODO: Add javadocs
  * TODO: Better naming
  */
-public class DisplayResolutionDependentFBOs extends AbstractFBOsManager {
+public class DisplayResolutionDependentFBOs extends AbstractFBOsManager implements PropertyChangeListener {
     public static final SimpleUri FINAL_BUFFER = new SimpleUri("engine:fbo.finalBuffer");
 
     private SwappableFBO gBufferPair;
 
-    private FBO.Dimensions fullScale;
+    private FBO.Dimensions fullScale = new FBO.Dimensions(-1, -1);
```

Comment:
The Dimensions class has no default constructor - meaning that I can
a) add a constructor for this particular case (meaning it's possible to instantiate dimensionless-dimensions),
or b) just set dimensions to -1 (or 0, since it's possible that Dimensions actually uses unsigned int internally) for a short while, till it gets properly initialized in the constructor.
I chose the latter :D

Reply:
Responded in the latest review.